feat: Support must_include and must_exclude regex as an array

bbugh opened the pull request:
Hi! :wave: Another PR. This solves our use case raised in #573 and allows `must_include` to be either a `string` or an `array.

I imagine this would be used in other places, but I'm not sure where.

shine2lay reviewed (approved):
Looks good, just for symmetry sake, would you mind doing the same for `must_exclude` as well? I think you also need to update the docs and changelog

bbugh commented:
Alright, I tested locally and it's working great. I extended it to include `must_exclude`, added some more tests, and updated the other validators that use `must_include` and `must_exclude` to support the array format. Also added the changelog. Let me know if there's anything else you want me to address!

bbugh commented:
Oh, also updated the docs for the `MustInclude` and `MustExclude` pages, though I'm not sure if I got everything.
